src/stan/services/util/run_sampler.hpp

```diff
@@ -45,7 +47,9 @@ void run_sampler(stan::mcmc::base_mcmc& sampler, Model& model,
                  size_t num_chains = 1) {
   Eigen::Map<Eigen::VectorXd> cont_params(cont_vector.data(),
                                           cont_vector.size());
-  services::util::mcmc_writer writer(sample_writer, diagnostic_writer, logger);
+  callbacks::structured_writer dummy_metric_writer;
+  services::util::mcmc_writer writer(sample_writer, diagnostic_writer,
+                                     dummy_metric_writer, logger);
```

Comment:
These kinds of changes are the thing we could avoid by adding another constructor to mcmc_writer

Reply:
not needed after all.